consensus/src/validator.rs

```diff
@@ -64,14 +65,22 @@ where
             block_index_handle,
         ),
         RequiredConsensus::IgnoreConsensus => validate_ignore_consensus(header.header()),
-        RequiredConsensus::PoS(pos_status) => validate_pos_consensus(
-            chain_config,
-            &pos_status,
-            block_index_handle,
-            utxos_view,
-            pos_accounting_view,
-            header,
-        ),
+        RequiredConsensus::PoS(pos_status) => {
+            // TODO: Remove the `pos_consensus_check` flag once PoS kernels can be checked normally.
+            // See also https://github.com/mintlayer/mintlayer-core/issues/752 for details.
+            if pos_consensus_check {
```

Comment:
Not sure if it makes sense but we could move this check inside `validate_pos_consensus` and disable checks inside partially. There are some usefull onces that don't access history, like target check and timestamp ordering. Maybe new name for this flag is required in that case.